data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java

```diff
@@ -176,16 +178,33 @@ public void visit(MessageVectorLayerData message) {
 
         @Override
         public void visit(MessageAlertData message) {
-            Alert alert = convertAlertData(message.getContent(), message.getMessageId());
-            mMessage = new MessageAlert(
-                    message.getMessageId(),
-                    message.getSenderId(),
-                    message.getCreatedAt(),
-                    alert);
+            if (message.getContent().location != null) {
+                GeoAlert alert = convertGeoAlertData(message.getContent(), message.getMessageId());
```

Comment:
Why do you pass the id parameter? In convertAlertData(AlertData content) you don't need it so why here? use content.messageId